test/from-draft3/to-draft4/required-is-present-in-outer-schema.json

```diff
@@ -0,0 +1,17 @@
+[
+    {
+        "title": "`required` is present in outer schema",
+        "from": {
+            "properties": {
+                "foo": {}
+            },
+            "required": false
+        },
+        "to": {
+            "properties": {
+                "foo": {}
+            },
+            "required": []
```

Comment:
Can we avoid an empty `required` and just remove it in these cases? Otherwise this is invalid in Draft 4, which dictates `required` cannot be empty: https://json-schema.org/draft-04/draft-fge-json-schema-validation-00#rfc.section.5.4.3

Reply:
Oh! correct, I should have another rule after the processing..?

Reply:
Yep, I believe so!